pkg/steps/index_generator_test.go

```diff
@@ -0,0 +1,77 @@
+package steps
+
+import (
+	"testing"
+
+	apiimagev1 "github.com/openshift/api/image/v1"
+	"github.com/openshift/ci-tools/pkg/api"
+	fakeimageclientset "github.com/openshift/client-go/image/clientset/versioned/fake"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestIndexGenDockerfile(t *testing.T) {
+	fakeClientSet := ciopTestingClient{
+		imagecs: fakeimageclientset.NewSimpleClientset(&apiimagev1.ImageStream{
+			ObjectMeta: v1.ObjectMeta{
+				Namespace: "target-namespace",
+				Name:      api.PipelineImageStream,
+			},
+			Status: apiimagev1.ImageStreamStatus{
+				PublicDockerImageRepository: "some-reg/target-namespace/stable",
+			},
+		}),
+		t: t,
+	}
+
+	var expectedDockerfileSingleBundle = `
+FROM quay.io/operator-framework/upstream-opm-builder AS builder
```

Comment:
Technically would be good to import these as `base_images` to the namespace so that they're considered when detecting the version of the job and so that we're not reliant on external registries/doing cluster-local pulls